src/components/Stream.tsx

```diff
@@ -32,13 +36,47 @@ export const Stream = () => {
   const { mediaStream } = useAppStream()
   const { overallState, immediateState } = useNetworkContext()
   const [streamState, setStreamState] = useState(StreamState.Unset)
+  const { file } = useRouteLoaderData(paths.FILE) as IndexLoaderData
 
   const IDLE = settings.context.app.streamIdleMode.current
 
   const isNetworkOkay =
     overallState === NetworkHealthState.Ok ||
     overallState === NetworkHealthState.Weak
 
+  /**
+   * Execute code and show a "building scene message"
+   * in Stream.tsx in the meantime.
+   *
+   * I would like for this to live somewhere more central,
+   * but it seems to me that we need the video element ref
+   * to be able to play the video after the code has been
+   * executed. If we can find a way to do this from a more
+   * central place, we can move this code there.
+   */
+  async function executeCodeAndPlayStream() {
+    kclManager.isFirstRender = true
+    kclManager.executeCode(true).then(() => {
+      videoRef.current?.play().catch((e) => {
+        console.warn('Video playing was prevented', e, videoRef.current)
+      })
+      kclManager.isFirstRender = false
+      setStreamState(StreamState.Playing)
+    })
+  }
+
+  /**
+   * Subscribe to execute code when the file changes
+   * but only if the scene is already ready.
+   * See onSceneReady for the initial scene setup.
+   */
+  useEffect(() => {
+    if (engineCommandManager.engineConnection?.isReady() && file?.path) {
+      console.log('execute on file change')
+      executeCodeAndPlayStream()
+    }
+  }, [file?.path])
```

Comment:
Does this need to also depend on the `engineConnection`'s readiness?

Reply:
Yeah I think adding `engineCommandManager.engineConnection` to the dependencies list would be a good idea here.